src/nashpy/linalg/tableau.py

```diff
@@ -0,0 +1,492 @@
+
+        Parameters
+        ----------
+        *kargs : Any
+            Positional arguments passed to Tableau constructor
+        **kwargs : Any
+            Key value arguments passed to Tableau constructor
+        """
+        self._non_basic_variables = None
+        super().__init__(*kargs, **kwargs)
+
+    @property
+    def non_basic_variables(self) -> Set:
+        """
+        Identifies the non basic variables of the tableau,
+        these correspond to the basic labels. For lex algorithm
+        a non-basic variable must have been dropped in an earlier pivot
+
+        Returns
+        -------
+        Set
+            The indices of the non basic variables.
+        """
+
+        if self._non_basic_variables is None:
+            self._non_basic_variables = super().non_basic_variables
+        return set(self._non_basic_variables)
+
+    def _find_pivot_row(self, column_index: int) -> int:
+        """
+        Finding the row to pivot like std tableau, but applying lex sorting to resolve ties in minratio.
+
+        Parameters
+        ----------
+        column_index : int
+            The column/label to pivot.
+
+        Returns
+        -------
+        int
+            The label to drop.
+        """
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                "ignore",
+                r"invalid value encountered in true_divide|divide by zero encountered in true_divide",
+            )
+            row_ratios = self._tableau[:, column_index] / self._tableau[:, -1]
+            row_ratios[np.isnan(row_ratios)] = -np.inf
+            ties = row_ratios == np.max(row_ratios)
+            if sum(ties) > 1:
+                return self._tie_break_lex(column_index, ties)
+            return int(np.argmax(row_ratios))
+
+    def _tie_break_lex(self, column_index: int, ties: npt.NDArray) -> int:
+        """
+        Provides tie breaks if minratio algorithm yields multiple rows. The tiebreaking looks at a perturbed
+        problem similar to https://people.math.carleton.ca/~kcheung/math/notes/MATH5801/05/5_3_perturb.html.
+
+        As tableau[:, slack_vars] started out as identity matrix then tableau[:, slack_vars] * e = e
+        tableau[:, slack_vars] * e should still reflect how pertubations were alterted through the pivoting matrix operations.
+        to tie break we apply minratio test on the pertubed problem. Rather than setting a value for the error, we simply
+        rely on error << min(tableau) so we can lexiographically compare entries in tableau
+
+        Parameters
+        ----------
+        column_index : int
+            The label to pivot.
+        ties: array
+            The rows to perform tiebreak on
+
+        Returns
+        -------
+        int
+            The row to pivot on
+        """
+        errs = self._tableau[:, sorted(self.slack_variables)]
+        pivot_column = self._tableau[:, (column_index,)]
+        err_ratios = errs / pivot_column
```

Comment:
If you compare with https://github.com/drvinceknight/Nashpy/pull/209/files#diff-afcbc2dd5fe0e1b292a6d2b26086885e45cbd461a859d4a32100904e20811b0eL50, this seems to be the fix for the bug. The original lex sort did not take into account the pivot column when lex sorting the perturbed tableau.

Beyond this had to add some handling for NaNs. Also struggled a bit getting test cases to complete. Basically its not entirely clear to me why we can do `pivot_col / tableau[:,-1]` in the max ratio check, while here its swapped.